### DISCUSSION TITLE
What's the best way to auto-prune only specific resources

### DISCUSSION THREAD

**Discussion by andrii-korotkov-verkada** [argoproj/argo-cd#18004] [Q&A]
I'm planning to migrate a service from HPA to KEDA ScaledObject and want to prune the HorizontalPodAutoscaler object after the ScaledObject is synced. Having auto-prune for the whole application is undesirable though. But manual sync with prune for each cluster is too tedious. I wonder if there's a way to selectively enable auto prune for specific resources or resource kinds.
One potential thing can be setting `http://argocd.argoproj.io/sync-options: Prune=true` before deleting the HPA resource, but I don't know if that's going to work. 
I'd appreciate the feedback and answers. Thank you.

**Comment by andrii-korotkov-verkada** [comment]
I've tried to add `http://argocd.argoproj.io/sync-options: Prune=true`, apply the changes and then do another apply to remove the resource. It didn't get pruned automatically.

**Comment by ChristianCiach** [comment]
Relevant issue: https://github.com/argoproj/argo-cd/issues/14277

Until this feature is added, I don't see another way than to do the inverse: Annotate all resources with `argocd.argoproj.io/sync-options: Prune=false` that you *don't want* to be pruned.
